integration/harness.go

```diff
@@ -87,3 +98,41 @@ func baseDir() (string, error) {
 	err := os.MkdirAll(baseDir, 0755)
 	return baseDir, err
 }
+
+func isListening(addr string) bool {
+	conn, _ := net.DialTimeout("tcp", addr, 1*time.Second)
+	if conn != nil {
+		_ = conn.Close()
+		return true
+	}
+	return false
+}
+
+func killProcess(address string) error {
+	addr, err := net.ResolveTCPAddr("tcp", address)
+	if err != nil {
+		return err
+	}
+
+	var cmd *exec.Cmd
+	if runtime.GOOS == "windows" {
+		args := fmt.Sprintf("(Get-NetTCPConnection -LocalPort %s).OwningProcess -Force", addr.Port)
+		cmd = exec.Command("Stop-Process", "-Id", args)
+	} else {
+		args := fmt.Sprintf("lsof -i tcp:%d | grep LISTEN | awk '{print $2}' | xargs kill -9", addr.Port)
```

Comment:
this is ubuntu specific commands, maybe ps is better?

Reply:
what do you mean by use go os? to inspect the platform?